include/rnp/rnp2.h

```diff
@@ -261,6 +265,57 @@ rnp_result_t rnp_key_have_public(rnp_key_handle_t key, bool *result);
 
 /* TODO define functions for password-based encryption */
 
+rnp_result_t rnp_op_sign_create(rnp_op_sign_t *op,
+                                rnp_ffi_t      ffi,
+                                rnp_input_t    input,
+                                rnp_output_t   output);
+
+rnp_result_t rnp_op_sign_add_signer(
+  rnp_op_sign_t    op,
+  rnp_key_handle_t key,
+  const char *     hash_fn,
+  uint32_t         creation_time, /* seconds since Jan 1 1970 UTC */
+  uint32_t         expiration_seconds);
+
+rnp_result_t rnp_op_sign_set_compression(rnp_op_sign_t op, const char *compression, int level);
+rnp_result_t rnp_op_sign_set_armor(rnp_op_sign_t op, bool armored);
+rnp_result_t rnp_op_sign_set_clearsign(rnp_op_sign_t op, bool clearsign);
```

Comment:
@ni4 is this what you had in mind here, or were you thinking a separate op (like `rnp_op_clearsign_*`) is best?
I can see an argument for having a separate op since clearsign+not-armored isn't a valid combo IIRC. Thoughts?

Reply:
Yeah, that's it - have clearsign/armor as separate functions. 
Creating another set of rnp_op_clearsign_* functions would be too redundant. 
Actually, once clearsign is set to true armoring flag doesn't make sense at all, so could be ignored (or asserted to report misuse).

Reply:
I agree. @randombit what do you think?